pkgs/top-level/stage.nix

```diff
@@ -176,155 +176,104 @@ let
       ((config.packageOverrides or (super: {})) super);
 
   # Convenience attributes for instantitating package sets. Each of
-  # these will instantiate a new version of allPackages. Currently the
-  # following package sets are provided:
-  #
-  # - pkgsCross.<system> where system is a member of lib.systems.examples
-  # - pkgsMusl
-  # - pkgsi686Linux
-  otherPackageSets = self: super: {
+  # these will instantiate a new version of allPackages.
+  otherPackageSets = let
+    mkPkgs = name: fn: nixpkgsFun (prevArgs: let nixpkgsArgs = fn prevArgs; in nixpkgsArgs // {
+      overlays = [
+        (self': super': {
+          "${name}" = super';
```

Comment:
Is this overlay still needed now that the package sets are composable?

It forces accessing an already accessed package set to be a no-op, which it now already is without the overlay for simple cases (e.g. `pkgsStatic.pkgsStatic`), and I believe is not desirable for more complex cases (e.g. when using `buildPackages`):

```nix
> pkgs.pkgsStatic.buildPackages.pkgsStatic.stdenv.hostPlatform.libc
"glibc"
```

Reply:
I've found that without the overlay above, using a package set in an overlay stack overflows:

```nix
> (pkgs.extend (final: prev: { inherit (final.pkgsStatic) hello; })).hello
error: stack overflow; max-call-depth exceeded
```

So a mechanism to avoid infinite recursion is indeed still needed, however I think the current one is not ideal, as it leads to unwanted no-ops (see the example in my previous comment).